### PR TITLE
Add to cart

### PR DESCRIPTION
**Pull request by sznowicki**
# Pull Request Template

## Description

Added a upselling widget within the Sheet which opens on `AddToCartSuccess` action.
Fixed small issues related to router problems.

## Type of change

- [x] Bug fix
- [x] New feature
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I updated the CHANGELOG.md

### CODE REVIEW

**Review by Carsten89 — COMMENTED** [review]
image

**Review by Carsten89 — COMMENTED** [review]
on small screens we should show only 2 items in a row i would say
image

**Review by Carsten89 — COMMENTED** [review]
you fixed already some ESlint errors in your last commit...but 3 are still left ;)

File: frontend/components/PDPSheet/index.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      // eslint-disable-next-line prefer-destructuring
+      routeDidChangeFunc = mockedRouteDidChangeSubscribe.mock.calls[0][0];
+      // eslint-disable-next-line prefer-destructuring
+      addToCartSuccessFunc = mockedPDPAddToCartSuccessSubscribe.mock.calls[0][0];
+    });
+
+    it('should react on add to cart', () => {
+      // Simulate add to cart event
+      addToCartSuccessFunc();
+      expect(instance.state.isOpen).toBe(true);
```
Comment (Carsten89) [inline frontend/components/PDPSheet/index.spec.jsx:131]: you should add a check here to test if the component really renders the correct content

File: frontend/components/PDPSheet/index.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    });
+
+    it('should react on route did change', () => {
+      // Simulate route did change.
+      routeDidChangeFunc();
+      expect(instance.state.isOpen).toBe(false);
```
Comment (Carsten89) [inline frontend/components/PDPSheet/index.spec.jsx:138]: you should add a check here to test if the component really renders the correct content

File: frontend/components/Grid/index.jsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import GridComponent from '@shopgate/pwa-common/components/Grid';
+import {
+  getProductRelationsFiltered,
+  getRelatedProductsByIdFiltered,
+} from '../../selectors';
+import styles from '../../styles/grid';
+import Item from '../Item';
+
+/**
+ * Slider component. Takes productId, type and additional props and renders Slider with real
```
Comment (Carsten89) [inline frontend/components/Grid/index.jsx]: copy paste ;)

**Review by Carsten89 — APPROVED** [review]

**Review by Carsten89 — CHANGES_REQUESTED** [review]
.

**Comment by sznowicki** [comment]
> on small screens we should show only 2 items in a row i would say
Your screen size is for devices like Nintendo DSi and HTC Wildfire A3333. For 320w which is something that we might actually have running the size in my opinion is still fine.

screenshot 2018-10-23 at 09 55 29

EDIT: convinced off-line, will prepare changes

File: frontend/components/PDPSheet/index.spec.jsx (hunk 1 above)
Comment (sznowicki) [reply frontend/components/PDPSheet/index.spec.jsx:131]: done

File: frontend/components/Grid/index.jsx (hunk 3 above)
Comment (sznowicki) [reply frontend/components/Grid/index.jsx]: done

File: frontend/components/PDPSheet/index.spec.jsx (hunk 2 above)
Comment (sznowicki) [reply frontend/components/PDPSheet/index.spec.jsx:138]: done

**Review by Carsten89 — APPROVED** [review]

**Pull request merged**